File: Project_Golang/ComputeHash.go
package main

import "crypto/sha256"

// "crypto/sha384"

/*
func ComputeHashCommon(AlgorithmIndex int, srcArray []byte) []byte {
	switch AlgorithmIndex {
	case 0:
		return md5.md5(srcArray)
	case 1:
		return sha1.Sum1(srcArray)
	case 2:
		return sha256.Sum256(srcArray)
	case 3:
	//   return sha384.Sum384(srcArray)
	case 4:
		return sha512.Sum512(srcArray)
	default:
		return sha256.Sum256(srcArray)
	}
}
*/
func ComputeHash_Common(AlgorithmIndex int, srcArray []byte) [32]byte {
	return sha256.Sum256(srcArray)
}

File: Project_Python3_NoCompare/mylibs/ComputeHash.py
# coding: utf-8

import hashlib

class ComputeHash:

    # 関数配列の呼び出し
    def ComputeHash_Common(self, i, srcArray):
        if i == 0:
            return hashlib.md5(srcArray).digest
        elif i == 1:
            return hashlib.sha1(srcArray).digest()
        elif i == 2:
            return hashlib.sha256(srcArray).digest()
        elif i == 3:
            return hashlib.sha384(srcArray).digest()
        elif i == 4:
            return hashlib.sha512(srcArray).digest()
        else:
            return hashlib.sha256(srcArray).digest()
